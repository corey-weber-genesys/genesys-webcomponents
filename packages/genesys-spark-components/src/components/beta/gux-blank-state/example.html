<h1>gux-blank-state-beta</h1>

<h4>
  The blank state will transition from large to medium or small when the view
  becomes smaller. Use the the drag icon to see the difference.
</h4>

<gux-blank-state-beta class="blank-state-styling">
  <gux-icon slot="image" icon-name="bot" decorative="true"></gux-icon>
  <div slot="primary-message">Sorry, something went wrong.</div>
  <div slot="additional-guidance">Please refresh this page to try again.</div>
  <button slot="call-to-action" type="button" onclick="notify(event)">
    Call to action
  </button>
</gux-blank-state-beta>

<style>
  .blank-state-styling {
    width: 250px;
    height: 250px;
    overflow: auto;
    resize: horizontal;
    border: 1px dashed gainsboro;
  }
</style>
