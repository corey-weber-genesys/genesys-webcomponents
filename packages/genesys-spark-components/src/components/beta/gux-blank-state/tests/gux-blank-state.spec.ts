jest.mock('../../../../utils/decorator/on-resize', () => ({
  __esModule: true,
  OnResize: jest.fn()
}));

import MutationObserver from 'mutation-observer';
import { newSpecPage } from '@stencil/core/testing';
import { GuxBlankState } from '../gux-blank-state';

const components = [GuxBlankState];
const language = 'en';

beforeEach(async () => {
  global.MutationObserver = MutationObserver;
});

describe('gux-blank-state-beta', () => {
  it('should build', async () => {
    const html = `
        <gux-blank-state-beta>
        <gux-icon slot="image" icon-name="bot" decorative="true"></gux-icon>
        <div slot="primary-message">Sorry, something went wrong.</div>
        <div slot="additional-guidance">Please refresh this page to try again.</div>
        <button slot="call-to-action" type="button" onclick="notify(event)">Call to action</button>
      </gux-blank-state-beta>`;

    const page = await newSpecPage({ components, language, html });
    const component = page.rootInstance;

    expect(component).toBeInstanceOf(GuxBlankState);
  });

  describe('#render', () => {
    it(`should render the component as expected`, async () => {
      const html = `
        <gux-blank-state-beta>
        <gux-icon slot="image" icon-name="bot" decorative="true"></gux-icon>
        <div slot="primary-message">Sorry, something went wrong.</div>
        <div slot="additional-guidance">Please refresh this page to try again.</div>
        <button slot="call-to-action" type="button" onclick="notify(event)">Call to action</button>
      </gux-blank-state-beta>`;
      const page = await newSpecPage({ components, language, html });

      expect(page.root).toMatchSnapshot();
    });
  });
});
