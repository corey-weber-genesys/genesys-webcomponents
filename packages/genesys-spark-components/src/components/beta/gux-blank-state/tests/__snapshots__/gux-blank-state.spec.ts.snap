// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-blank-state-beta #render should render the component as expected 1`] = `
<gux-blank-state-beta>
  <mock:shadow-root>
    <div class="gux-container gux-large">
      <slot name="image"></slot>
      <div class="gux-message-container">
        <slot name="primary-message"></slot>
      </div>
      <div class="gux-guidance-container">
        <slot name="additional-guidance"></slot>
      </div>
      <gux-button-slot-beta accent="primary">
        <slot name="call-to-action"></slot>
      </gux-button-slot-beta>
    </div>
  </mock:shadow-root>
  <gux-icon decorative="true" icon-name="bot" slot="image"></gux-icon>
  <div slot="primary-message">
    Sorry, something went wrong.
  </div>
  <div slot="additional-guidance">
    Please refresh this page to try again.
  </div>
  <button onclick="notify(event)" slot="call-to-action" type="button">
    Call to action
  </button>
</gux-blank-state-beta>
`;
